File: SMSscanorm3/SMSscanorm3.R
rm(list=ls(all=TRUE))
graphics.off()

# install and load packages
libraries = c("MASS", "scatterplot3d")
lapply(libraries, function(x) if (!(x %in% installed.packages())) {
  install.packages(x)
})
lapply(libraries, library, quietly = TRUE, character.only = TRUE)

set.seed(100);                             # pseudorandom numbers

n     = 300;                               # number of observations
mu    = c(0,0,0);                          # mean
sigma = diag(x = 1, nrow = 3, ncol = 3);   # covariance matrix

x     = mvrnorm(n, mu, sigma);             # multivariate normal random numbers

scatterplot3d(x, color = "red", main = "Standard Normal point cloud", grid = FALSE,xlab="",ylab="",zlab="");

File: SMSscanorm3/SMSscanorm3.m
clear all
close all
clc

% set pseudo random numbers
RandStream.setGlobalStream(RandStream('mt19937ar','seed',98));

% main calculation
n = 300;            % number of observations
mu = [0,0,0];      % mean
sigma = diag([1,1,1]); % standard deviation

x = mvnrnd(mu, sigma,n); % generation of multivariate normal random numbers

% scatterplot

scatter3(x(:,1),x(:,2),x(:,3),'r','LineWidth',2);
xlim([-3,3]);
ylim([-3,3]);
zlim([-3,3]);
title('Standard Normal point cloud','FontSize',16,'FontWeight','Bold');
set(gca,'LineWidth',1.6,'FontSize',16,'FontWeight','Bold');


% to save plots please uncomment following lines
% print -painters -dpdf -r600 SMSscanorm3.pdf
% print -painters -dpng -r600 SMSscanorm3.png
